scripts/tests/twister_blackbox/test_error.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python3
+# Copyright (c) 2024 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+"""
+Blackbox tests for twister's command line functions - simple does-error-out or not tests
+"""
+
+import importlib
+import mock
+import os
+import pytest
+import sys
+
+from conftest import ZEPHYR_BASE, TEST_DATA, testsuite_filename_mock
+from twisterlib.testplan import TestPlan
+from twisterlib.error import TwisterRuntimeError
+
+
+class TestError:
+    TESTDATA_1 = [
+        (
+            os.path.join('scripts', 'tests', 'twister_blackbox', 'test_data', 'tests',
+                         'dummy', 'agnostic', 'group1', 'subgroup1',
+                         'dummy.agnostic.group1.subgroup1'),
+            SystemExit
+        ),
+        ('dummy.agnostic.group1.subgroup1', TwisterRuntimeError),
+    ]
+
+    @classmethod
+    def setup_class(cls):
+        apath = os.path.join(ZEPHYR_BASE, 'scripts', 'twister')
+        cls.loader = importlib.machinery.SourceFileLoader('__main__', apath)
+        cls.spec = importlib.util.spec_from_loader(cls.loader.name, cls.loader)
+        cls.twister_module = importlib.util.module_from_spec(cls.spec)
+
+    @classmethod
+    def teardown_class(cls):
+        pass
+
+    @pytest.mark.usefixtures("clear_log")
+    @pytest.mark.parametrize(
+        'test, expected_exception',
+        TESTDATA_1,
+        ids=['valid', 'invalid']
+    )
+    @mock.patch.object(TestPlan, 'TESTSUITE_FILENAME', testsuite_filename_mock)
+    def test_test(self, out_path, test, expected_exception):
```

Comment:
can we add fucntion doc string here to explain  details of test approach? it is not friendly to see code to undersand the logic.

Reply:
I think that such a comment shouldn't be in a specific test_ file, as it is information pertinent to any blackbox test. Where should be put such information then? In conftest? In a test_dummy.py, which would hold a single, highly commented testcase?

Reply:
> I think that such a comment shouldn't be in a specific test_ file, as it is information pertinent to any blackbox test. Where should be put such information then? In conftest? In a test_dummy.py, which would hold a single, highly commented testcase?

`./scripts/tests/twister_blackbox/README.rst` ?

Reply:
I will create a separate PR for a readme file.